explorer/redis.go

```diff
@@ -0,0 +1,37 @@
+package explorer
+
+import (
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+// SetRedisKey to Get conn and put back when exit from method
+func (a *App) SetRedisKey(key string, val []byte, expiration uint64) error {
+	conn := a.redis.Get()
+	defer conn.Close()
+
+	_, err := conn.Do("SET", key, val, "EX", expiration)
+	if err != nil {
+		log.Error().Err(errors.Wrap(err, fmt.Sprintf("ERROR: fail set key %s, val %s", key, val))).Msg("")
```

Comment:
As  a general rule Logging should only be done in the caller. Instead this method should return error directly **after** it's wrapper in proper message
```
return errors.Wrapf(err, "failed to set key %s, val %s", key, value)
// or more stanard way
return fmt.Errorf("failed to set key blah blah : %w", err) // notice it's %w so it does wrapping of the cause
```

Note: the log can stay here if needed, but the error always need to be wrapper. u see now the error u return doesn't tell what operation was going on when u got this redis error